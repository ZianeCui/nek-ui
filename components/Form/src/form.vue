<template>
  <div>
    <form action="">
      <input type="text" v-model="inputValue">
      <span class="reversed">{{ reversedInput }}</span>
    </form>
  </div>
</template>

<script>
export default {
  props: ['reversed'],
  data: () => ({
    inputValue: ''
  }),
  computed: {
    reversedInput() {
      return this.reversed ?
        this.inputValue.split("").reverse().join("") :
        this.inputValue
    }
  },
  watch: {
    inputValue(newVal, oldVal) {
      if(newVal.trim().length && newVal !== oldVal) {
        console.log(newVal)
      }
    }
  }
}
</script>