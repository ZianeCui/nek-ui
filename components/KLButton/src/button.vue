<template>
    <button :disabled="disabled" class="kl-button" type="button" @click="handleClick">
        {{ title }}
    </button>
</template>

<script>
export default {
    name: 'KLButton',
    props: {
        disabled: Boolean
    }
};
</script>

<style lang="scss">
.kl-button {
    // - 移除浏览器中的margin。| Remove margins in Chrome, Safari and Opera.
    -webkit-appearance: none;
    margin: 0;
    // - 移除浏览器中的默认border。| Remove borders for `button`.
    overflow: visible;
    // - 让字体继承父节点的。| Correct `font` properties not being inherited for `button`.
    font: inherit;
    // - Address inconsistent `text-transform` inheritance which is only inherit in Firefox and IE.
    text-transform: none;
    // - 为了`a`标签。| Required for `a`.
    text-decoration: none;
    // - 统一用`pointer`，在`a`标签没有`href`的时候鼠标指针不是pointer。 | Use pointer cursor at any time.
    cursor: pointer;
    // - 统一box-sizing，a标签默认的box-sizing是content-box，而button标签默认的box-sizing是border-box。
    box-sizing: border-box;
    // - Reset unusual Firefox-on-Android default style; see https://github.com/necolas/normalize.css/issues/214
    background: none;
    // - Others
    display: inline-block;
    vertical-align: middle;
    text-align: center;
    height: 24px;
    min-width: 80;
    padding: 0 16;
    background: #fff;
    border: 1px solid #ddd;
    border-radius: 2px;
    line-height: 22;
    font-size: 12px;
    color: #333;
    transition-duration: 0.1s;
    transition-property: transform;
    transition-timing-function: ease;
    white-space: nowrap;
}
/* Hover */
.kl-button:hover,
.kl-button:focus { // - Apply hover style also to focus state
    // - Remove default focus style
    outline: none;
    // - Required for `a` elements
    text-decoration: none;
    background: #fff;
    border-color: #bbb;
}
/* Active */
.kl-button:active,
.kl-button.z-act {
    background: #fff;
}
</style>

