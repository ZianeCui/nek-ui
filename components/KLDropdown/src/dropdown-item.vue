<template>
    <div class="kl-dropdown-item" v-on:click="handleClick">
        <slot></slot>
    </div>
</template>

<script>
    export default {
        name: 'kl-dropdown-item',
        props: {

        },
        methods: {
            handleClick: function() {
                console.log(this)
                this.$emit('click', this);
            }
        },
        computed: {

        }
    };
</script>

<style lang="scss">
    .kl-dropdown-item {
        margin: 0;
        line-height: normal;
        padding: 7px 16px;
        clear: both;
        color: #495060;
        font-size: 12px!important;
        white-space: nowrap;
        list-style: none;
        cursor: pointer;
        transition: background .2s ease-in-out;
    }
    .kl-dropdown-item:hover {
        background: #f3f3f3;
    }
</style>