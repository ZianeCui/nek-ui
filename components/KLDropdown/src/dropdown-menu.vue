<template>
    <div class="kl-dropdown-menu"><slot></slot></div>
</template>
<script>
    export default {
        name: 'kl-dropdown-menu'
    };
</script>

<style lang="scss">
    .kl-dropdown-menu {
        min-width: 100px;
    }
</style>