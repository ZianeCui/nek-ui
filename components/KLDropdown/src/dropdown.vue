<template>
    <div class="kl-dropdown">
        <div class="kl-dropdown-header" @click="onClick">
            <slot></slot>
        </div>
        <div class="kl-dropdown-footer" v-bind:class="{ 'kl-dropdown-footer-show': visible }">
            <slot name="list"></slot>
        </div>
    </div>
</template>
<script>
    export default {
        name: 'kl-dropdown',
        props: {
            // visible: false
        },
        computed: {

        },
        watch: {

        },
        methods: {
            onClick: function() {
                console.log(this)
                this.visible = true;
                this.$emit('click');
            }
        },
        data: function () {
            return {
               visible: false
            }
        }
    };
</script>
<style lang="scss">
    .kl-dropdown {
        display: inline-block;
        position: relative;
        .kl-dropdown-header {
            display: inline-block;
            font-size: 12px;
        }
        .kl-dropdown-footer {
            display: none;
            position: absolute;
            width: inherit;
            max-height: 200px;
            overflow: auto;
            margin: 5px 0;
            padding: 5px 0;
            background-color: #fff;
            box-sizing: border-box;
            border-radius: 4px;
            box-shadow: 0 1px 6px rgba(0,0,0,.2);
            position: absolute;
            z-index: 900;
        }
        .kl-dropdown-footer-show {
            display: inline-block;
        }
    }
</style>