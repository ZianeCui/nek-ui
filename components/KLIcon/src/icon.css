@font-face {
    font-family: 'icomoon';
    src:  url('iconfonts/icomoon.eot?7jwxq3');
    src:  url('iconfonts/icomoon.eot?7jwxq3#iefix') format('embedded-opentype'),
      url('iconfonts/icomoon.ttf?7jwxq3') format('truetype'),
      url('iconfonts/icomoon.woff?7jwxq3') format('woff'),
      url('iconfonts/icomoon.svg?7jwxq3#icomoon') format('svg');
    font-weight: normal;
    font-style: normal;
  }

  [class^="kl-icon-"], [class*=" kl-icon-"] {
    /* use !important to prevent issues with browser extensions that change fonts */
    font-family: 'icomoon' !important;
    speak: none;
    font-style: normal;
    font-weight: normal;
    font-variant: normal;
    text-transform: none;
    line-height: 1;

    /* Better Font Rendering =========== */
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  .kl-icon-rotate_left:before {
    content: "\e900";
  }
  .kl-icon-upload:before {
    content: "\e901";
  }
  .kl-icon-download:before {
    content: "\e902";
  }
  .kl-icon-zoomin:before {
    content: "\e903";
  }
  .kl-icon-expand:before {
    content: "\e904";
  }
  .kl-icon-cycle_delete:before {
    content: "\e905";
  }
  .kl-icon-zoomout:before {
    content: "\e906";
  }
  .kl-icon-rezoom:before {
    content: "\e907";
  }
  .kl-icon-rotate_right:before {
    content: "\e908";
  }
  .kl-icon-retry:before {
    content: "\e909";
  }
  .kl-icon-warning:before {
    content: "\e90a";
  }
  .kl-icon-edit:before {
    content: "\e90b";
  }
  .kl-icon-info:before {
    content: "\e90c";
  }
  .kl-icon-error:before {
    content: "\e90d";
  }
  .kl-icon-success:before {
    content: "\e90e";
  }
  .kl-icon-chevron_left:before {
    content: "\e90f";
  }
  .kl-icon-chevron_right:before {
    content: "\e910";
  }
  .kl-icon-search:before {
    content: "\e911";
  }
  .kl-icon-remove:before {
    content: "\e912";
  }
  .kl-icon-ok:before {
    content: "\e913";
  }
  .kl-icon-check_empty:before {
    content: "\e914";
  }
  .kl-icon-angle_down:before {
    content: "\e915";
  }
  .kl-icon-cycle_add:before {
    content: "\e916";
  }
  .kl-icon-calendar:before {
    content: "\e91b";
  }
  .kl-icon-line:before {
    content: "\e91c";
  }
  .kl-icon-trash:before {
    content: "\e91f";
  }
  .kl-icon-plus:before {
    content: "\e920";
  }
  .kl-icon-home:before {
    content: "\e921";
  }
