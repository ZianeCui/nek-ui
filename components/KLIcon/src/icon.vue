<template>
    <i class="kl-icon" :class="klass" :style="style" @click="onClick"></i>
</template>

<script>
export default {
    name: 'kl-icon',
    props: {
        type: String,
        fontSize: String,
        color: String,
    },
    methods: {
        onClick() {
            this.$emit('click');
        }
    },
    computed: {
        klass () {
            return {
                [`kl-icon-${this.type}`]: !!this.type,
            }
        },
        style () {
            let style = {};

            if (!!this.fontSize) {
                style.fontsize = this.fontSize + 'px';
            }

             if (!!this.color) {
                style.color = this.color;
            }

            return style;
        }
    }
};
</script>

<style lang="scss">
    @import 'icon.css'
</style>

