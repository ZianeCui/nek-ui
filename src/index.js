import KLButton from '../components/KLButton/index';
import KLIcon from '../components/KLIcon/index';
import KLInput from '../components/KLInput/index';
import KLDropdown from '../components/KLDropdown/index';
import KLDropdownMenu from '../components/KLDropdownMenu/index';
import KLDropdownItem from '../components/KLDropdownItem/index';
import KLMessage from '../components/KLMessage/index';
import Form from '../components/Form/index';
import List from '../components/List/index';
import ListItem from '../components/ListItem/index';

import MessageList from '../components/Example/eg5/MessageList/index'
import Message from '../components/Example/eg5/Message/index'

module.exports = {
    KLButton,
    KLIcon,
    KLInput,
    KLDropdown,
    KLDropdownMenu,
    KLDropdownItem,
    KLMessage,
    Form,
    List,
    ListItem,
    MessageList,
    Message,
}