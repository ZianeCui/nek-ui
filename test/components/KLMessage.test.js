import Vue from 'vue'
import { KLMessage } from '../../src/index'
import { shallow, mount } from 'vue-test-utils'

import KLMessageItem from '../../components/KLMessage/src/message-item.vue'


describe('KLMessage.test.js', () => {
    let cmp, vm
    let test2

    beforeEach(() => {
        cmp = shallow(KLMessage, {
            data: {
                messages: ['a', 'b', 'c']
            }
        })
    })

    it('equals messages to ["a", "b", "c"]', () => {
        expect(cmp.vm.messages).toEqual(['a', 'b', 'c'])
    })

    it('has the expected html structure1', () => {
        expect(cmp.vm.$el).toMatchSnapshot()
    })




    let cmp2 = mount(KLMessage)

    it('contains a MessageItem component', () => {
        expect(cmp2.contains(KLMessageItem)).toBe(true)
    })


    it('Calls handleMessageClick when @message-click happens', () => {
        const stub = jest.fn()
        cmp2.setMethods({ handleMessageClick: stub })
        cmp2.update()

        const el = cmp2.find(KLMessageItem).vm.$emit('message-clicked', 'cat')
        expect(stub).toBeCalledWith('cat')
      })



})