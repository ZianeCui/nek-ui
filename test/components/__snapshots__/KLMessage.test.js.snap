// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KLMessage.test.js has the expected html structure1 1`] = `
<div
  id="app"
>
  <!---->
  <!---->
  <!---->
</div>
`;
